src/include/txn_inline.h

```diff
@@ -929,7 +929,11 @@ __wt_upd_alloc(WT_SESSION_IMPL *session, const WT_ITEM *value, u_int modify_type
         upd->size = WT_STORE_SIZE(value->size);
         memcpy(upd->data, value->data, value->size);
     }
-    upd->type = (uint8_t)modify_type;
+    /*
+     * This field is const but we need to set it once at allocation time, to do so temporarily cast
+     * as a non-const.
+     */
+    *(uint8_t *)&upd->type = (uint8_t)modify_type;
```

Comment:
This is really weird. Can we change the code to not initialize this variable before and initialize it here so we don't have to do the cast? Otherwise, I would prefer not to declare it as const.

Reply:
The structure is calloc'd so it performs the initialization in that function, we are writing to it the second time in C's opinion here. Which is why we cast the const away. The other option is to create a struct, locally then memcpy it to the new struct which is calloc'd to get around the const. Either way I do not consider removing the const an option as it provides a guarantee that we need to uphold in WiredTiger, and it does so at compile time.

Reply:
I would replace calloc with malloc and manually initialize all the variable.

Reply:
Theres a good explanation [here](https://stackoverflow.com/questions/9691404/how-to-initialize-const-in-a-struct-in-c-with-malloc).

Reply:
Changing the calloc to a malloc still has the same problem as it is an assignment to a read only variable.